cmd/ido2icingadb/convert.go

```diff
@@ -0,0 +1,816 @@
+package main
+
+import (
+	"database/sql"
+	"github.com/icinga/icingadb/pkg/contracts"
+	v1 "github.com/icinga/icingadb/pkg/icingadb/v1"
+	"github.com/icinga/icingadb/pkg/icingadb/v1/history"
+	icingadbTypes "github.com/icinga/icingadb/pkg/types"
+	"github.com/icinga/icingadb/pkg/utils"
+	"github.com/jmoiron/sqlx"
+	"github.com/pkg/errors"
+	"strings"
+	"time"
+)
+
+const acknowledgementMigrationQuery = "SELECT ah.acknowledgement_id, UNIX_TIMESTAMP(ah.entry_time) entry_time, " +
+	"ah.entry_time_usec, ah.acknowledgement_type, ah.author_name, ah.comment_data, ah.is_sticky, " +
+	"ah.persistent_comment, UNIX_TIMESTAMP(ah.end_time) end_time, o.objecttype_id, o.name1, " +
+	"IFNULL(o.name2, '') name2 " +
+	"FROM icinga_acknowledgements ah USE INDEX (PRIMARY) " +
+	"INNER JOIN icinga_objects o ON o.object_id=ah.object_id " +
+	"WHERE ah.acknowledgement_id > ? ORDER BY ah.acknowledgement_id LIMIT ?"
+
+type AckClear struct {
+	Id        icingadbTypes.Binary
+	ClearTime icingadbTypes.UnixMilli
+}
+
+var _ contracts.TableNamer = (*AckClear)(nil)
+
+func (*AckClear) TableName() string {
+	return "acknowledgement_history"
+}
+
+func convertAcknowledgementRows(
+	env string, envId, endpointId icingadbTypes.Binary,
+	selectCache func(dest interface{}, query string, args ...interface{}), _ *sqlx.Tx, idoRows []struct {
+		AcknowledgementId   uint64
+		EntryTime           int64
+		EntryTimeUsec       uint32
+		AcknowledgementType uint8
+		AuthorName          sql.NullString
+		CommentData         sql.NullString
+		IsSticky            uint8
+		PersistentComment   uint8
+		EndTime             sql.NullInt64
+		ObjecttypeId        uint8
+		Name1               string
+		Name2               string
+	},
+) (icingaDbUpdates, icingaDbInserts [][]interface{}, checkpoint interface{}) {
```

Comment:
Does this really need to define full struct types within a function parameter? I think it would be easier to read if this was just a regular named type.

Reply:
Well, you usually give a type a name if you wanna equip it with methods or re-use it. Both are not the case here.

Reply:
Or if defining it inline makes things hard to read.